src/cpio/client_providers/blob_storage_client_provider/azure/azure_blob_storage_client_provider.cc

```diff
@@ -0,0 +1,122 @@
+#include "src/core/utils/base64.h"
+#include "src/core/utils/hashing.h"
+#include "src/cpio/client_providers/blob_storage_client_provider/common/error_codes.h"
+#include "src/public/core/interface/execution_result.h"
+#include "src/public/cpio/interface/blob_storage_client/type_def.h"
+
+using google::cmrt::sdk::blob_storage_service::v1::Blob;
+using google::cmrt::sdk::blob_storage_service::v1::BlobMetadata;
+using google::cmrt::sdk::blob_storage_service::v1::DeleteBlobRequest;
+using google::cmrt::sdk::blob_storage_service::v1::DeleteBlobResponse;
+using google::cmrt::sdk::blob_storage_service::v1::GetBlobRequest;
+using google::cmrt::sdk::blob_storage_service::v1::GetBlobResponse;
+using google::cmrt::sdk::blob_storage_service::v1::GetBlobStreamRequest;
+using google::cmrt::sdk::blob_storage_service::v1::GetBlobStreamResponse;
+using google::cmrt::sdk::blob_storage_service::v1::ListBlobsMetadataRequest;
+using google::cmrt::sdk::blob_storage_service::v1::ListBlobsMetadataResponse;
+using google::cmrt::sdk::blob_storage_service::v1::PutBlobRequest;
+using google::cmrt::sdk::blob_storage_service::v1::PutBlobResponse;
+using google::cmrt::sdk::blob_storage_service::v1::PutBlobStreamRequest;
+using google::cmrt::sdk::blob_storage_service::v1::PutBlobStreamResponse;
+using google::protobuf::util::TimeUtil;
+using google::scp::core::AsyncContext;
+using google::scp::core::AsyncPriority;
+using google::scp::core::ConsumerStreamingContext;
+using google::scp::core::ExecutionResult;
+using google::scp::core::ExecutionResultOr;
+using google::scp::core::FailureExecutionResult;
+using google::scp::core::FinishContext;
+using google::scp::core::ProducerStreamingContext;
+using google::scp::core::RetryExecutionResult;
+using google::scp::core::SuccessExecutionResult;
+using google::scp::core::common::kZeroUuid;
+using google::scp::core::common::TimeProvider;
+using google::scp::core::errors::SC_BLOB_STORAGE_PROVIDER_ERROR_GETTING_BLOB;
+using google::scp::core::errors::SC_BLOB_STORAGE_PROVIDER_INVALID_ARGS;
+using google::scp::core::errors::SC_BLOB_STORAGE_PROVIDER_RETRIABLE_ERROR;
+using google::scp::core::errors::
+    SC_BLOB_STORAGE_PROVIDER_STREAM_SESSION_CANCELLED;
+using google::scp::core::errors::
+    SC_BLOB_STORAGE_PROVIDER_STREAM_SESSION_EXPIRED;
+using google::scp::core::utils::Base64Encode;
+
+namespace google::scp::cpio::client_providers {
+
+absl::Status AzureBlobStorageClientProvider::GetBlob(
+    AsyncContext<GetBlobRequest, GetBlobResponse>& get_blob_context) noexcept {
+  return absl::OkStatus();
+}
+
+absl::Status AzureBlobStorageClientProvider::GetBlobStream(
+    ConsumerStreamingContext<GetBlobStreamRequest, GetBlobStreamResponse>&
+        get_blob_stream_context) noexcept {
+  return absl::OkStatus();
+}
+
+absl::Status AzureBlobStorageClientProvider::ListBlobsMetadata(
+    AsyncContext<ListBlobsMetadataRequest, ListBlobsMetadataResponse>&
+        list_blobs_context) noexcept {
+  return absl::OkStatus();
```

Comment:
Should we implement something to indicate this is not implemented?

Reply:
It's fine in a sense that we will clarify that's not implemented yet in the PR (and we did so in previous PRs), but agree that we can improve it.
Adding a comment `// Not implemented.` seems to be what upstream does like src/cpio/client_providers/instance_client_provider/test_instance_client_provider.cc.